unity-renderer/Assets/Scripts/MainScripts/DCL/Components/Avatar/AvatarShape.cs

```diff
@@ -256,8 +254,24 @@ public void SetImpostor(string userId)
             currentLazyObserver.AddListener(avatar.SetImpostorTexture);
         }
 
-        private void PlayerPointerExit() { playerName?.SetForceShow(false); }
-        private void PlayerPointerEnter() { playerName?.SetForceShow(true); }
+        private void PlayerPointerExit()
+        {
+            DataStore.i.outliner.avatarOutlined.Set((null, -1, -1));
+            playerName?.SetForceShow(false);
+        }
+
+        private void PlayerPointerEnter()
```

Comment:
I have a feeling this stuff should be moved out of `AvatarShape`, it is too m uch for a single class. But I see that it follows the existing pattern so probably it can be addressed separately unless you have a fast idea where it fits more properly

Reply:
There's a new implementation for AvatarShape in ECS7.Altough both follow a similar pattern, this one will be deprecated "soon-ish".

I agree though, AvatarShape is a mess.